### PR TITLE
Makes python-3-conversion branch pass all build tests again

### PR DESCRIPTION
**Pull request by JossWhittle**
PR remade specifically from commit `c96c19b`

During the last couple of months the default python3 pickle version has changed. This has caused files created by the Opt-ID to have slightly different header lengths and values than those originally created by python3 cPickle that are used to validate test results. 

This PR implements `Magnets::__eq__` and `MagLists::__eq__` deep comparison methods to compare that two python class instances really do contain the same data values. The test cases have been updates to load in expected and observed files using pickle and then offloading the comparison to those methods. 

Additionally this PR includes minor quality of life and typo fixes in debug messages, and tries to apply the use of `with open() as file:` closures where needed to ensure that files are closed gracefully when exceptions are thrown.

### CODE REVIEW

File: IDSort/src/magnets.py [hunk 1]
```diff
@@ -32,19 +32,20 @@ def __init__(self):
...
+            for line in fp:
+                vals = line.split()
+                assert len(vals) >= 4
+                magnets[vals[0]] = np.array((float(vals[1]), float(vals[2]), float(vals[3])))
+
         self.magnet_sets[name] = magnets
         self.magnet_flip[name] = np.array(flip_vector)
-        
-        self.mean_field[name]=0.0
+        self.mean_field[name]  = 0.0
```
Comment (markbasham) [inline IDSort/src/magnets.py:45]: Double space here.

File: IDSort/src/magnets.py [hunk 2]
```diff
@@ -62,31 +63,64 @@ def add_perfect_magnet_set_duplicate(self, name, mag_list, vector, flip_vector):
...
+        return True
 
 class MagLists():
     '''
     This class deals with the ordering and specification of several lists
     '''
     
     def __init__(self, magnets):
-        self.raw_magnets = magnets
+        self.raw_magnets  = magnets
```
Comment (markbasham) [inline IDSort/src/magnets.py:123]: Double space here

File: IDSort/src/magnets.py [hunk 3]
```diff
@@ -172,46 +226,15 @@ def process(options, args):
     if options.htmags:
         mags.add_magnet_set('HT', options.htmags, (-1.,-1.,1.))
     
-    #mags.add_perfect_magnet_set('HH', 20 , (0.,0.,1.), (-1.,1.,-1.))
```
Comment (markbasham) [inline IDSort/src/magnets.py:175]: Is this section not important, or is it just testing?

File: IDSort/src/process_genome.py [hunk 4]
```diff
@@ -13,178 +13,113 @@
...
-                
-                line= ("%5i %4i %4i %4i %4i %03i\n"%(b+1,a+1,mag_type,info['beams'][b]['mags'][a]['direction_matrix'][2][2],mag_flip, mag_num))
-                f3.write(line)
-                
-                a+=1
-                
-            f3.write("\n")
-        
-        f3.close()
+    if info['type'] == 'Hybrid_Symmetric':
```
Comment (markbasham) [inline IDSort/src/process_genome.py:32]: Like this :) much more pythonic

File: IDSort/src/magnets.py (hunk 3 above)
Comment (JossWhittle) [reply IDSort/src/magnets.py:175]: It appeared to all be for testing. Apart from these mags.add_perfect calls everything modifies the MagList instance and then the original unmodified Magnets instances is what is saved. I think this was some experimentation to soft validate the classes were working. 

I'd like to eventually have some unit tests that do the same sort of things.

File: IDSort/src/process_genome.py (hunk 4 above)
Comment (JossWhittle) [reply IDSort/src/process_genome.py:32]: This is where I was referring to about having test data for the other ID types. I've checked carefully the refactor is doing the same thing as the original would have, the only real differences are the presence of the HT beam and magnet direction (matrix). 

Need to have a chat at some point about the naming conventions for the different beams just so I can picture where they are placed w.r.t one another in an ID.

File: IDSort/src/magnets.py (hunk 3 above)
Comment (markbasham) [reply IDSort/src/magnets.py:175]: Ok cool, thanks

**Comment by JossWhittle** [comment]
First PR of many to come :)

**Pull request merged**